### PR TITLE
Insert options filtered by exists and appending first object 

### PR DESCRIPTION
**Pull request by gayanDeegala**

### CODE REVIEW

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx [hunk 1]
```diff
@@ -13,7 +13,9 @@ const cretaValidInsertMenu = (submenu: MenuItem[] | undefined, validInsertItems:
     submenu?.forEach(subItem => {
         if (validInsertItems !== undefined && validInsertItems.length !== 0) {
             Object.keys(validInsertItems).forEach((key: any) => {
-                if (subItem.text === key && subItem.title === validInsertItems[key].description) {
+                if (!existingKeys.includes(key) &&
```
Comment (malindakpt) [inline src/components/JsonEditorContainer/JsonEditorContainer.tsx:16]: Add validation for non-unique desc

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx [hunk 2]
```diff
@@ -34,14 +46,23 @@ export function JsonEditorContainer(props: { json: any, templates: Template[] })
             console.log(err.toString())
         },
         onCreateMenu: (menuItems: MenuItem[], node: MenuItemNode) => {
-            const paths = node.paths[0];
+            // get current state
+            let currentJson = jsonEditorInstance.current?.getText();
+            if (currentJson) {
```
Comment (malindakpt) [inline src/components/JsonEditorContainer/JsonEditorContainer.tsx:51]: Do not use the same variable for different purposes

**Review by malindakpt — APPROVED** [review]

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx (hunk 2 above)
Comment (gayanDeegala) [reply src/components/JsonEditorContainer/JsonEditorContainer.tsx]: Done

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx (hunk 1 above)
Comment (gayanDeegala) [reply src/components/JsonEditorContainer/JsonEditorContainer.tsx:18]: added that

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx [hunk 3]
```diff
@@ -12,17 +12,36 @@ const cretaValidInsertMenu = (submenu: MenuItem[] | undefined, validInsertItems:
...
+                if (!existingKeys.includes(key) &&
+                    subItem.text === key &&
+                    subItem.title === validInsertItems[key].description) {
                     validMenuItems.push(subItem);
+                    matchingCount++;
                 }
             });
+
+            if (matchingCount > 1) {
+                console.error("non-unique Menu Items, Please use the valid option");
```
Comment (malindakpt) [inline src/components/JsonEditorContainer/JsonEditorContainer.tsx:27]: replace console.log with alert()

Comment (gayanDeegala) [reply src/components/JsonEditorContainer/JsonEditorContainer.tsx:27]: sure

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx [hunk 4]
```diff
@@ -12,17 +12,36 @@ const cretaValidInsertMenu = (submenu: MenuItem[] | undefined, validInsertItems:
 
     submenu?.forEach(subItem => {
         if (validInsertItems !== undefined && validInsertItems.length !== 0) {
+            let matchingCount = 0;
```
Comment (malindakpt) [inline src/components/JsonEditorContainer/JsonEditorContainer.tsx:15]: matchingItemCountWithSameDesc

**Review by malindakpt — APPROVED** [review]

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx (hunk 4 above)
Comment (gayanDeegala) [reply src/components/JsonEditorContainer/JsonEditorContainer.tsx]: Done

**Pull request merged**